pkgs/development/libraries/xeus-zmq/default.nix

```diff
@@ -0,0 +1,38 @@
+{ clangStdenv
+, cmake
+, fetchFromGitHub
+, lib
+
+# Dependency libraries
+, cppzmq
+, libuuid
+, nlohmann_json
+, openssl
+, xeus
+, xtl
+, zeromq
+}:
+
+clangStdenv.mkDerivation rec {
+  pname = "xeus-zmq";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "jupyter-xeus";
+    repo = "xeus-zmq";
+    rev = "${version}";
+    sha256 = "0m9g0fqm001d0jk91yk139d42azcinzhkqhaw7qhx3dh18zcsvcg";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ cppzmq openssl libuuid xeus xtl zeromq ];
+  propagatedBuildInputs = [ nlohmann_json ];
+
+  meta = with lib; {
+    description = "ZeroMQ-based middleware for xeus";
+    homepage = "https://github.com/jupyter-xeus/xeus-zmq";
+    maintainers = with maintainers; [ thomasjm ];
+    platforms = platforms.all;
+    license = licenses.bsd3;
+  };
```

Comment:
```suggestion
  meta = {
    description = "ZeroMQ-based middleware for xeus";
    homepage = "https://github.com/jupyter-xeus/xeus-zmq";
    license = lib.licenses.bsd3;
    maintainers = with lib.maintainers; [ thomasjm ];
    platforms = lib.platforms.all;
  };
```

Reply:
Same question again -- I've gotten the opposite feedback in a prior review, to use a `with let` block. Doing some grepping, the way I've done it here seems to be almost universally how the rest of Nixpkgs does it.

Reply:
This is a bit more technical:

`with lib` acts like `#include <stdlib>` in C, bringing all elements of it into the scope and introducing complications on static analysis and many other things.
Also, the nesting of `with`s brings some nasty inconsistencies.

https://nix.dev/recipes/best-practices#with-scopes

https://github.com/NixOS/nixpkgs/issues/208242

Reply:
Both are fine and using with lib here is fine. There is no need to change packages over and it is definitely not something blocking a merge.